### PR TITLE
Add sort for react non paginated list

### PR DESCRIPTION
**Pull request by Tcharl**
I struggled a bit generating a sample project.
Also started to split small things that will prepare the sort for non paginated element in react

Contributes to https://github.com/jhipster/generator-jhipster/issues/18778
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
I don’t know any other node package that have this behavior. IMO we should drop this.

`jhipster` for global and `npx jhipster` for local is the normal behavior.
And v8 global jhipster command will fail to invoke local v7. And global v7 will fail to invoke local v8.

**Comment by Tcharl** [comment]
Still, the only way to make it work...
I also started another small thing on it...

**Comment by Tcharl** [comment]
Can it be reviewed and then https://github.com/jhipster/react-jhipster/pull/89 merged so that I'll update that branch to be merged too?
I'll continue the sort stuff and try to export everything I can to the lib.
I would like to do it by baby step instead of big bang thing (I already failed twice implementing it)

**Comment by Tcharl** [comment]
@qmonmert ?

**Comment by Tcharl** [comment]
Hmmmm, I think I made a mistake, I merged react-jhipster after the review, but now I'm unable to build the branch.
It's unfortunately a breaking change, as the `getSortState`is replaced by `getPaginationState`. @all, really sorry for that: what is the procedure to fix it?

**Comment by mshima** [comment]
@Tcharl no problem.
Just change the react-jhipster to the git main branch.

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "react": "<%= this.dependabotPackageJson.dependencies['react'] %>",
     "react-dom": "<%= this.dependabotPackageJson.dependencies['react-dom'] %>",
     "react-hook-form": "<%= this.dependabotPackageJson.dependencies['react-hook-form'] %>",
-    "react-jhipster": "<%= this.dependabotPackageJson.dependencies['react-jhipster'] %>",
+    "react-jhipster": "jhipster/react-jhipster",
```
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:44]: change at package.json instead of the template file

Comment (Tcharl) [reply generators/client/templates/react/package.json.ejs]: Ahhhh, thanks!

**Pull request merged**